### PR TITLE
Improve presentation of graphical view table in disk management

### PR DESCRIPTION
**Pull request by lukaszgo1**
### Link to issue number:
None
### Summary of the issue:
#9118 introduced support for graphical view of partition in Windows disk management. While it worked well for the most part UX could be better in some places namely:
1. Role of the partitions and disk was either table cell or table row header.
2. 'selected' was announced with every object.
3. When braille display was used and guessing of object position was enabled the reported positions were insane such as 65537 of 12.
4. Tool tips appearing when mouse was moved to one of these objects were reported as empty. Additionally when last object of the table was reached and mouse was positioned on one of the partitions empty tooltipp was announced over and over with each arrow press.
### Description of how this pull request fixes the issue:
1. Now these controls are reclassed to list item, which in addition to giving more user-friendly role suppresses announcement of selected.
2.  Reporting of position info is disabled for the graphical view table-  see known issues below.
3. Display mode is used to retrieve text of tooltips and repeated one are filtered out.
### Testing performed:
On Windows 7 and Windows 10 1903:
1. Moved to the table with TAB, made sure that selected isn't announced when navigating and role is list item.
2. With guessing of position info enabled and disabled ensured that there is no difference in output with both speech and braille.
3. With reporting of tooltips enabled ensured that they are reported and the repeated one are filtered out.
### Known issues with pull request:
- For some tool tips show event never fires. I have no idea why and honestly don't think that it is particularly important.
- I've disabled reporting of position info because I have no idea how to reasonably present current position in the four directional and irregular table such as this one. It should be however possible to calculate correct values to some extend, because while the coordinates given were incorrect, they were incorrect in a regular way.

### Change log entry:

Section: changes
- Presentation of graphical view table in disk management has been improved.

### CODE REVIEW

**Comment by lukaszgo1** [comment]
@codeofdusk You may be interested in taking a look at this.

File: source/appModules/mmc.py [hunk 1]
```diff
@@ -20,10 +22,36 @@ def _get_focusRedirect(self):
...
+	of suppressing selected. """
+	role = controlTypes.ROLE_LISTITEM
+
 	def event_selection(self):
 		if self.parent.hasFocus and api.getFocusObject() != self:
 			eventHandler.executeEvent("gainFocus", self)
 
+	def _get_positionInfo(self):
+		""" When 'Guess object position info when unavailable' is enabled
+		these controls reports very strange information such as 65537 of 12, especially in braille.
```
Comment (leonardder) [inline source/appModules/mmc.py]:
```suggestion
		these controls report very strange information such as 65537 of 12, especially in braille.
```

File: source/appModules/mmc.py [hunk 2]
```diff
@@ -20,10 +22,36 @@ def _get_focusRedirect(self):
...
+		""" ToolTips appearing when hovering mouse over graphical view have empty name,
+		but it can be retrieved via display model. """
+		return self.displayText
+
+	def event_show(self):
+		# Stop repeating the same tooltip over and over again.
+		toolTipText = self.displayText
+		if toolTipText != emptyToolTip.previousToolTipText:
+			emptyToolTip.previousToolTipText = toolTipText
+			return super(emptyToolTip, self).event_show()
```
Comment (leonardder) [inline source/appModules/mmc.py]: Events don't return anything. Also, notice the new, easier super call that is supported in Python 3.

```suggestion
			super().event_show()
```

File: source/appModules/mmc.py [hunk 3]
```diff
@@ -20,10 +22,36 @@ def _get_focusRedirect(self):
...
+		return None
```
Comment (leonardder) [inline source/appModules/mmc.py]:
```suggestion
```

File: source/appModules/mmc.py [hunk 4]
```diff
@@ -20,10 +22,36 @@ def _get_focusRedirect(self):
...
 			eventHandler.executeEvent("gainFocus", self)
 
+	def _get_positionInfo(self):
+		""" When 'Guess object position info when unavailable' is enabled
+		these controls reports very strange information such as 65537 of 12, especially in braille.
+		Disable reporting of position info all together. """
+		return None
+
+
+class emptyToolTip(ToolTip):
```
Comment (leonardder) [inline source/appModules/mmc.py]: You name this EmptyTooltip, but the code suggests that this tooltip can contain text. Therefore EmptyTooltip doesn't seem appropriate here.

**Review by LeonarddeR — DISMISSED** [review]

**Comment by lukaszgo1** [comment]
@leonardder All done.

**Comment by lukaszgo1** [comment]
@leonardder This is ready for another review when you have time.

**Comment by lukaszgo1** [comment]
@leonardder You have a pending review here.

**Comment by codeofdusk** [comment]
Looks okay to me.

**Review by feerrenrut — DISMISSED** [review]
This looks fine to me, I'll target this to 2020.4.

**Comment by feerrenrut** [comment]
Please could you make a suggestion for a change log entry (or several). Despite this being a cosmetic change, it is helpful for people to know it has been improved, and for testers to know what to focus on.

**Comment by lukaszgo1** [comment]
Hi @feerrenrut . Assuming that you're happy with these changes  could you please merge this? I certainly don't want to rash you, but given for how long this has been open it'd be a pity to miss 2020.4 for this one.

**Comment by dingpengyu** [comment]
Are there any known problems with current PR?
I hope to see this PR in NVDA2020.4.

**Review by feerrenrut — DISMISSED** [review]
Thanks @lukaszgo1

**Review by feerrenrut — APPROVED** [review]

**Pull request merged**